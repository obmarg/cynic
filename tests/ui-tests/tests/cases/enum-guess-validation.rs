fn main() {}

mod schema {
    cynic::use_schema!("../../../cynic/src/bin/simple.graphql");
}

#[derive(cynic::Enum, Clone, Copy, Debug)]
#[cynic(
    schema_path = "../../../cynic/src/bin/simple.graphql",
    query_module = "schema",
    graphql_type = "Desseqt"
)]
enum Dessert {
    CHEESECAKE,
    ICE_CREAM,
}
