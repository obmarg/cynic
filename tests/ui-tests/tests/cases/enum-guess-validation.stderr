error: Could not find an enum named Desseqt in ../../../cynic/src/bin/simple.graphql. Did you mean `Dessert`?
  --> tests/cases/enum-guess-validation.rs:10:5
   |
10 |     graphql_type = "Desseqt"
   |     ^^^^^^^^^^^^

warning: variant `ICE_CREAM` should have an upper camel case name
  --> tests/cases/enum-guess-validation.rs:14:5
   |
14 |     ICE_CREAM,
   |     ^^^^^^^^^ help: convert the identifier to upper camel case: `IceCream`
   |
   = note: `#[warn(non_camel_case_types)]` on by default
